source/CIS/cis-4.rst

```diff
@@ -0,0 +1,705 @@
+
+  CredentialStatus ::= (0: Byte) // Active
+                     | (1: Byte) // Revoked
+                     | (2: Byte) // Expired
+                     | (3: Byte) // NotActivated
+
+.. _CIS-4-functions:
+
+Contract functions
+------------------
+
+A smart contract implementing this standard MUST export the following functions:
+
+- :ref:`CIS-4-functions-credentialEntry`
+- :ref:`CIS-4-functions-credentialStatus`
+- :ref:`CIS-4-functions-issuer`
+- :ref:`CIS-4-functions-issuerMetadata`
+- :ref:`CIS-4-functions-registerCredential`
+- :ref:`CIS-4-functions-revokeCredentialIssuer`
+- :ref:`CIS-4-functions-revokeCredentialHolder`
+- :ref:`CIS-4-functions-revokeCredentialOther`
+- :ref:`CIS-4-functions-registerRevocationKeys`
+- :ref:`CIS-4-functions-removeRevocationKeys`
+- :ref:`CIS-4-functions-revocationKeys`
+
+
+.. _CIS-4-functions-credentialEntry:
+
+``credentialEntry``
+^^^^^^^^^^^^^^^^^^^
+
+Query a credential entry from the registry by ID.
+
+Parameter
+~~~~~~~~~
+
+The parameter is the credential ID.
+
+See the serialization rules in :ref:`CIS-4-CredentialID`.
+
+Response
+~~~~~~~~
+
+The function returns a registry entry corresponding to the credential ID parameter.
+
+It is serialized as :ref:`CIS-4-CredentialInfo` (``credential_info``) followed by a credential schema reference :ref:`CIS-4-SchemaRef` (``schema_ref``), and a credential entry revocation nonce :ref:`CIS-4-Nonce` (``revocation_nonce``)::
+
+  CredentialQueryResponse ::= (credential_info: CredentialInfo) (schema_ref: SchemaRef) (revocation_nonce: Nonce)
+
+
+Requirements
+~~~~~~~~~~~~
+
+- The query MUST fail if the credential ID is unknown.
+
+.. _CIS-4-functions-credentialStatus:
+
+``credentialStatus``
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+Query the status of a credential from the credential registry by ID.
+
+Parameter
+~~~~~~~~~
+
+The parameter is the credential ID.
+
+See the serialization rules in :ref:`CIS-4-CredentialID`.
+
+Response
+~~~~~~~~
+
+The function returns the status of a credential.
+
+See the serialization rules in :ref:`CIS-4-CredentialStatus`
+
+Requirements
+~~~~~~~~~~~~
+
+- The query MUST fail if the credential ID is unknown.
+- The credential status MUST be ``Expired`` if the credential is not revoked, the field ``valid_until`` was present in :ref:`CIS-4-CredentialInfo` when registering the credential, and ``valid_until < now``.
+- The credential status MUST NOT be ``Expired`` if the field ``valid_until`` was not present in :ref:`CIS-4-CredentialInfo` when registering the credential.
+- The credential status MUST be ``Acive`` if the credential is not revoked, and does not qualify as ``Expired`` or ``NotActivated``.
+
+.. _CIS-4-functions-issuer:
+
+``issuer``
+^^^^^^^^^^
+
+Query the issuer's account address.
+
+Response
+~~~~~~~~
+
+The function output is the issuer's account address.
+It is serialized as :ref:`CIS-4-AccountAddress`.
+
+.. _CIS-4-functions-issuerMetadata:
+
+``issuerMetadata``
```

Comment:
This entrypoint doesn't have a parameter. This is metadata for the whole register, since it's a single issuer per register.
(There is a different kind of metadata: credential metadata. To query that one you need a credential id)

Reply:
Ah, my mistake.